cmd/maya-exporter/app/collector/cstorcollector_test.go

```diff
@@ -0,0 +1,190 @@
+package collector
+
+import (
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"regexp"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+var (
+	Listener net.Listener
+	resp1    = "iSCSI Target Controller version istgt:0.5.20121028:20:26:33:May 21 2018 on  from\r\nIOSTATS  IQN=iqn.2017-08.OpenEBS.cstor:vol1 Blockcount=20971520 Blocklength=512 Writes=0 Reads=0 TotalReadBytes=0 TotalWriteBytes=0 Size=10737418240\r\nOK IOSTATS\r\n"
+	resp2    = "IOSTATS  IQN=iqn.2017-08.OpenEBS.cstor:vol1 Blockcount=20971520 Blocklength=512 Writes=0 Reads=0 TotalReadBytes=0 TotalWriteBytes=0 Size=10737418240\r\nOK IOSTATS\r\n"
+)
+
+func TestSplit(t *testing.T) {
+	cases := map[string]struct {
+		response string
+		output   map[string]string
+	}{
+		"Response with header": {
+			response: resp1,
+			output: map[string]string{
+				"TotalWriteBytes": "0",
+				"Size":            "10737418240",
+				"IQN":             "iqn.2017-08.OpenEBS.cstor:vol1",
+				"Blockcount":      "20971520",
+				"Blocklength":     "512",
+				"Writes":          "0",
+				"Reads":           "0",
+				"TotalReadBytes":  "0",
+			},
+		},
+		"Response without header": {
+			response: resp2,
+			output: map[string]string{
+				"TotalWriteBytes": "0",
+				"Size":            "10737418240",
+				"IQN":             "iqn.2017-08.OpenEBS.cstor:vol1",
+				"Blockcount":      "20971520",
+				"Blocklength":     "512",
+				"Writes":          "0",
+				"Reads":           "0",
+				"TotalReadBytes":  "0",
+			},
+		},
+	}
+	for name, tt := range cases {
+		t.Run(name, func(t *testing.T) {
+			got := split(tt.response)
+			if len(got) != len(tt.output) {
+				t.Fatalf("split(%v) : expected %v, got %v", tt.response, got, tt.output)
+			}
+		})
+	}
+}
+
+func runFakeUnixServer(t *testing.T, wg *sync.WaitGroup) {
+	go func() {
+		// start the server
+		//var err error
+		t.Log("server started")
+		listener, err := net.Listen("unix", "@anysocketpath")
+		wg.Done()
+		if err != nil {
+			t.Fatal(err)
+		}
+		for {
+			fd, err := listener.Accept()
+			if err != nil {
+				t.Fatal("Accept error: ", err)
+			}
+			go sendFakeResponse(t, fd)
+		}
+	}()
+}
+
+func sendFakeResponse(t *testing.T, c net.Conn) {
+	for {
+		buf := make([]byte, 512)
+		_, err := c.Read(buf)
+		if err != nil {
+			return
+		}
+
+		data := resp1
+		t.Logf("Received: %v", data)
+		_, err = c.Write([]byte(data))
+		if err != nil {
+			panic("Write: " + err.Error())
+		}
+	}
+}
+
+// TestCollector tests cstorcollector.go
+func TestCstorCollector(t *testing.T) {
+
+	for _, tt := range []struct {
+		input string
+		match []*regexp.Regexp
+	}{
+		// this is the input we are passing for positive testing
+		// match will expect similar output from response.
+		{
+			input: `
+{
+	"stats": {
```

Comment:
It is little bit difficult to understand the test logic. I am able to make it out though.
However, imagine the cases where new scenarios need to be added or something breaks due to logic change.
Can the test case be made more simpler?

Reply:
i think i need some help here to make it simpler, perhaps you can suggest some ways to make it simpler and concise.